### DISCUSSION TITLE
Error in installing foundry devops .

### DISCUSSION THREAD

**Discussion by khushbusandhu** [Cyfrin/foundry-full-course-cu#641] [General]
got this error while installing foundry-devops
```
Installing foundry-devops in /home/khushbu/foundry-smart-contract-lottery-f23/lib/foundry-devops (url: Some("https://github.com/ChainAccelOrg/foundry-devops"), tag: None)
Cloning into '/home/khushbu/foundry-smart-contract-lottery-f23/lib/foundry-devops'...
fatal: unable to access 'https://github.com/ChainAccelOrg/foundry-devops/': Could not resolve host: github.com
fatal: clone of 'https://github.com/ChainAccelOrg/foundry-devops' into submodule path '/home/khushbu/foundry-smart-contract-lottery-f23/lib/foundry-devops' failed
Error: 
git submodule exited with code 128

```

**Comment by theirrationalone** [comment]
Please use...

```terminal
forge install cyfrin/foundry-devops --no-commit
```

**Reply by khushbusandhu** [reply]
It is also giving the same error.

**Reply by khushbusandhu** [reply]
Actually there is no  .gitmodule file is there in files so I guess this error is there

**Reply by theirrationalone** [reply]
but that should be created created on `forge init` automatically....

Let's experiment...

Try to create `.gitmodule` and add this into it...

```terminal
[submodule "lib/foundry-devops"]
	path = lib/foundry-devops
	url = https://github.com/Cyfrin/foundry-devops
```
Next, Install again with this more accurate command...

```terminal
forge install cyfrin/foundry-devops@0.0.11 --no-commit
```

**Reply by khushbusandhu** [reply]
getting the same error

**Reply by theirrationalone** [reply]
Hey, I just noticed this log...

```log
unable to access 'https://github.com/ChainAccelOrg/foundry-devops/': Could not resolve host: github.com
```

Seems like you're not loggedIn to your github account or Internet Connection* Lost????

**Reply by khushbusandhu** [reply]
no, I am logged in to GitHub and I have a decent internet connection too.

**Reply by theirrationalone** [reply]
are you using wsl???

**Reply by khushbusandhu** [reply]
this is the error 
```
Installing foundry-devops in /home/khushbu/foundry-smart-contract-lottery-f23/lib/foundry-devops (url: Some("https://github.com/cyfrin/foundry-devops"), tag: None)
Cloning into '/home/khushbu/foundry-smart-contract-lottery-f23/lib/foundry-devops'...
fatal: unable to access 'https://github.com/cyfrin/foundry-devops/': Could not resolve host: github.com
fatal: clone of 'https://github.com/cyfrin/foundry-devops' into submodule path '/home/khushbu/foundry-smart-contract-lottery-f23/lib/foundry-devops' failed
Error: 
git submodule exited with code 128
```

**Reply by khushbusandhu** [reply]
yes i am using wsl

**Reply by theirrationalone** [reply]
plzz show me the command you're executing...

**Reply by khushbusandhu** [reply]
here's the command
```
forge install cyfrin/foundry-devops@0.0.11 --no-commit
```

**Reply by theirrationalone** [reply]
would you like to share your's vscode's screen cap???

**Reply by khushbusandhu** [reply]
how to share

**Reply by theirrationalone** [reply]
Hit `windows + PtrSc` when vscode's window is open, -> then go to Pictures -> Screenshots -> your captured SS.

**Reply by khushbusandhu** [reply]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/121751731/410dfdf1-2194-47c3-8a51-b98893a99850)

**Reply by theirrationalone** [reply]
Please check git.user & git.email.....??

**Reply by khushbusandhu** [reply]
same error again

**Reply by theirrationalone** [reply]
Sorry my bad try this... there's a capital `C` not `c` in cyfrin. 😅

```terminal
forge install Cyfrin/foundry-devops --no-commit
```

**Reply by khushbusandhu** [reply]
same error

**Reply by theirrationalone** [reply]
Let's Experiment one more thing... 

make a `.gitmodule` file and put statements given below and then try again... 

```git
 [submodule "lib/chainlink-brownie-contracts"] 
         path = lib/chainlink-brownie-contracts 
         url = https://github.com/smartcontractkit/chainlink-brownie-contracts 
 [submodule "lib/forge-std"] 
         path = lib/forge-std 
         url = https://github.com/foundry-rs/forge-std
```

**Reply by shikhar229169** [reply]
I think it is due to network issues, no doubt you are able to browse normally but there is network issue in wsl.
Try to restart vs code.

Try downgrading wsl 2 to wsl 1, it worked for my friend, he was having the same network issue in wsl.

**Reply by khushbusandhu** [reply]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/121751731/02c72b2a-6a9c-4dc6-ab39-ee9ae1b8c24d)

**Reply by khushbusandhu** [reply]
Thanks alot @theirrationalone and @shikhar229169  it worked after restarting vs code 😅

**Reply by theirrationalone** [reply]
It's definately permissions issue Or github might gone down, but we can chat still so that's not a problem. 

Please check your user.name & user.email
```terminal
git config user.name
git config user.email
```

**Reply by khushbusandhu** [reply]
maybe vs code have some problem, I restarted it and then the same command worked

**Reply by theirrationalone** [reply]
Wowww @shikhar229169 ;) .... 

It might be firewall under the hood??

**Reply by khushbusandhu** [reply]
yup! You guessed it right @theirrationalone 😂

**Reply by theirrationalone** [reply]
Heyy @khushbusandhu, 
These all consequences sounds like we should start learning linux a bit.... 😅😅

**Reply by khushbusandhu** [reply]
yes I also think so 😅
Heyy @shikhar229169  would you like to teach about Linux to me and @theirrationalone 😂

**Reply by theirrationalone** [reply]
Yes please..... @shikhar229169 ! We'll be thankful.

**Reply by shikhar229169** [reply]
I don't have much depth knowledge of it, idk I do random stuffs and it worked. You won't believe my linux  was creating the same network problem, I restarted it several times no luck and at last I went to chatgpt and it said to restart it and guess what it worked.
